src/basic/terminal-util.c

```diff
@@ -708,6 +709,57 @@ char *resolve_dev_console(char **active) {
         return tty;
 }
 
+int get_kernel_consoles(char ***consoles) {
+        _cleanup_strv_free_ char **con = NULL;
+        _cleanup_free_ char *active = NULL;
+        const char *tty, *state;
+        size_t len;
+        int ret;
+
+        assert(consoles);
+
+        ret = read_one_line_file("/sys/class/tty/console/active", &active);
+        if (ret < 0)
+                return ret;
+
+        FOREACH_WORD(tty, len, active, state) {
+                char *path;
+
+                if (strneq(tty, "tty0", 4)) {
```

Comment:
This is wrong, because it only compares the first four characters, even if `tty` is longer. It probably doesn't matter in practice too much, but still, it should be fixed. When you convert to `extract_first_word`, the issue should go away.